src/network.test.ts

```diff
@@ -0,0 +1,50 @@
+import Dockerode from "dockerode";
+import { GenericContainer } from "./generic-container";
+import { Network } from "./network";
+
+jest.setTimeout(45000);
+
+describe("Network", () => {
+  it("should start container in a user-defined network", async () => {
+    const network = await Network.newNetwork();
+
+    const container = await new GenericContainer("cristianrgreco/testcontainer", "1.1.12")
+      .withNetworkMode(network.getName())
+      .start();
+
+    const dockerodeClient = new Dockerode();
+
+    const dockerContainer = dockerodeClient.getContainer(container.getId());
+    const containerInfo = await dockerContainer.inspect();
+    expect(containerInfo.HostConfig.NetworkMode).toBe(network.getName());
+
+    await container.stop();
+    await network.close();
+  });
+
+  it("two containers in user-defined network should be able to ping each other by name", async () => {
+    const network = await Network.newNetwork();
+
+    const container1 = await new GenericContainer("cristianrgreco/testcontainer", "1.1.12")
+      .withName("container1")
+      .withNetworkMode(network.getName())
+      .start();
+
+    const container2 = await new GenericContainer("cristianrgreco/testcontainer", "1.1.12")
+      .withName("container2")
+      .withNetworkMode(network.getName())
+      .start();
+
+    const { exitCode } = await container1.exec(["ping", "-c", "3", "container2"]);
+
+    expect(exitCode).toBe(0);
+
+    await container1.stop();
+    await container2.stop();
+    await network.close();
+  });
+
+  it("instantiating Network should throw", () => {
+    expect(() => new Network()).toThrowError("use static newNetwork() method to instantiate network");
```

Comment:
Why not use the constructor instead of having an additional method?

Reply:
That's how it was implemented in `testcontainers-java` - i thought we were meant to take inspiration from there.